public/app/features/dashboard-scene/sharing/components/ImagePreview.tsx

```diff
@@ -0,0 +1,119 @@
+import { useMemo, useEffect } from 'react';
+import { useMeasure } from 'react-use';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { t } from '@grafana/i18n';
+import { Alert, LoadingBar, Text, useStyles2 } from '@grafana/ui';
+
+type ErrorState = {
+  message: string;
+  title: string;
+  code?: string;
+} | null;
+
+interface ImagePreviewProps {
+  imageBlob: Blob | null;
+  isLoading: boolean;
+  error: ErrorState;
+  testId?: string;
+  title?: string;
+  showLoading?: boolean;
+}
+
+export function ImagePreview({
+  imageBlob,
+  isLoading,
+  error,
+  testId = selectors.components.ExportImage.preview.container,
+  title,
+  showLoading = true,
+}: ImagePreviewProps) {
+  const styles = useStyles2(getStyles);
+  const [ref, { width: measuredWidth }] = useMeasure<HTMLDivElement>();
+
+  // Memoize and clean up the object URL for the image
+  const imageUrl = useMemo(() => {
+    if (!imageBlob) {
+      return undefined;
+    }
+    const url = URL.createObjectURL(imageBlob);
+    return url;
+  }, [imageBlob]);
+
+  useEffect(() => {
+    return () => {
+      if (imageUrl) {
+        URL.revokeObjectURL(imageUrl);
+      }
+    };
+  }, [imageUrl]);
+
+  return (
+    <div className={styles.previewContainer} ref={ref} data-testid={testId}>
+      {showLoading && isLoading && (
+        <div className={styles.loadingBarContainer} data-testid={selectors.components.ExportImage.preview.loading}>
+          <LoadingBar width={measuredWidth} />
+          {title && (
+            <div className={styles.titleContainer}>
+              <Text variant="body">{title}</Text>
+            </div>
+          )}
+        </div>
+      )}
+
+      {error && !isLoading && <ErrorAlert error={error} />}
+      {imageBlob && !isLoading && imageUrl && (
+        <img
+          src={imageUrl}
+          alt={t('share-modal.image.preview', 'Preview')}
+          className={styles.image}
+          data-testid={selectors.components.ExportImage.preview.image}
+          aria-label={t('share-modal.image.preview-aria', 'Generated image preview')}
+        />
+      )}
+    </div>
+  );
+}
+
+function ErrorAlert({ error }: { error: ErrorState }) {
+  if (!error) {
+    return null;
+  }
+
+  return (
+    <Alert severity="error" title={error.title} data-testid={selectors.components.ExportImage.preview.error.container}>
+      <div data-testid={selectors.components.ExportImage.preview.error.title}>{error.title}</div>
+      <div data-testid={selectors.components.ExportImage.preview.error.message}>{error.message}</div>
+    </Alert>
+  );
+}
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  previewContainer: css({
+    position: 'relative',
+    width: '100%',
+    minHeight: '200px',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: theme.colors.background.secondary,
+    borderRadius: theme.shape.radius.default,
+    overflow: 'hidden',
```

Comment:
is this container necessary?
Also, when there's an error it places an alert with this big box background 
<img width="764" alt="image" src="https://github.com/user-attachments/assets/e29359f2-ed24-40a6-8ef4-a709624e15e6" />

Reply:
container is a phantom loading container for when image is being generated - i think having error show up in same area makes sense for now 🤷